naga/src/front/wgsl/parse/directive/language_extension.rs

```diff
@@ -60,9 +60,25 @@ impl LanguageExtension {
     }
 }
 
-/// A variant of [`LanguageExtension::Implemented`].
+// A variant of [`LanguageExtension::Implemented`].
+/// Enum of all WGSL extensions that are implemented
+/// WGSL spec.: <https://www.w3.org/TR/WGSL/#language-extensions-sec>
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-pub(crate) enum ImplementedLanguageExtension {}
+pub enum ImplementedLanguageExtension {}
+
+impl ImplementedLanguageExtension {
+    const ALL: [ImplementedLanguageExtension; 0] = [];
+
+    /// Returns slice of all variants of [`ImplementedLanguageExtension`]
```

Comment:
style: Missing a period for what is otherwise a complete sentence.

```suggestion
    /// Returns slice of all variants of [`ImplementedLanguageExtension`].
```